types/src/block_metadata.rs

```diff
@@ -3,9 +3,22 @@ use crypto::{ed25519::Ed25519Signature, HashValue};
 use std::collections::BTreeMap;
 
 /// Struct that will be persisted on chain to store the information of the current block.
+///
+/// The flow will look like following:
+/// 1. The executor will pass this struct to VM at the end of a block proposal.
+/// 2. The VM will use this struct to create a special system transaction that will modify the on
+///    chain resource that represents the information of the current block. This transaction can't
+///    be emitted by regular users and won't be sequenced by the consensus. It's generated by each
```

Comment:
Instead of saying that it "won't be sequenced by consensus", perhaps say "will be executed before all of the user-submitted transactions in the block"?